src/main/java/ecomod/common/tiles/TileAdvancedFilter.java

```diff
@@ -13,30 +9,27 @@
 import ecomod.common.tiles.compat.CommonCapsWorker;
 import ecomod.common.utils.EMUtils;
 import ecomod.core.EMConsts;
-import ecomod.core.EcologyMod;
 import ecomod.core.stuff.EMConfig;
 import ecomod.core.stuff.EMIntermod;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.ITickable;
 import net.minecraft.util.SoundCategory;
-import net.minecraft.util.SoundEvent;
 import net.minecraft.util.math.MathHelper;
 import net.minecraftforge.common.capabilities.Capability;
-import net.minecraftforge.fluids.Fluid;
 import net.minecraftforge.fluids.FluidStack;
 import net.minecraftforge.fluids.FluidTank;
 import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
-import net.minecraftforge.fml.common.Optional;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
-public class TileAdvancedFilter extends TileEnergy implements ITickable, IHasWork
+@net.minecraftforge.fml.common.Optional.Interface(iface = "buildcraft.api.tiles.IHasWork", modid = "buildcraft")
```

Comment:
I don't think that this Optional.Interface annotation is useful because the intention is to handle IHasWork tiles even in absence of BuildCraft.

Reply:
The reason I added it was it was causing crashes without having BuildCraft installed and why would there be a BuildCraft IHasWork instance without BuildCraft

Reply:
Oh, sorry😓! I didn't notice then that the annotation is attached to TileEntity and it's not used in work check algorithms.